data-prepper-plugin-framework/src/main/java/org/opensearch/dataprepper/plugin/ExtensionLoader.java

```diff
@@ -62,6 +82,8 @@ public List<? extends ExtensionPlugin> loadExtensions() {
                         return null;
                     }
                 })
+                .sorted(Comparator.comparing(ExtensionPluginWithContext::isConfigured).reversed())
```

Comment:
It would be ideal to have this `Comparator` in another class so that it can grow over time without having to change this method. And it could have it's own unit tests.

You could create a named bean in `PluginCreatorContext`.

```
@Bean(name = "extensionsLoaderComparator")
public Comparator<ExtensionLoader.ExtensionPluginWithContext> extensionsLoaderComparator() {
  return Comparator.comparing(ExtensionPluginWithContext::isConfigured).reversed();
}
```